modules/auxiliary/gather/get_user_spns.py

```diff
@@ -0,0 +1,320 @@
+#!/usr/bin/env python2.7
+# -*- coding: utf-8 -*-
+
+import sys
+import os
+from datetime import datetime
+from binascii import hexlify, unhexlify
+
+# extra modules
+dependencies_missing = False
+try:
+    from pyasn1.codec.der import decoder
+    from impacket import version
+    from impacket.dcerpc.v5.samr import UF_ACCOUNTDISABLE, UF_NORMAL_ACCOUNT
+    from impacket.examples import logger
+    from impacket.krb5 import constants
+    from impacket.krb5.asn1 import TGS_REP
+    from impacket.krb5.ccache import CCache
+    from impacket.krb5.kerberosv5 import getKerberosTGT, getKerberosTGS
+    from impacket.krb5.types import Principal
+    from impacket.ldap import ldap, ldapasn1
+    from impacket.smbconnection import SMBConnection
+    from impacket.ntlm import compute_lmhash, compute_nthash
+except ImportError:
+    dependencies_missing = True
+
+from metasploit import module
+
+metadata = {
+    'name': 'Gather Ticket Granting Service (TGS) tickets for User Service Principal Names (SPN)',
+    'description': '''
+        This module will try to find Service Principal Names that are associated with normal user accounts.
+        Since normal accounts' passwords tend to be shorter than machine accounts, and knowing that a TGS request
+        will encrypt the ticket with the account the SPN is running under, this could be used for an offline
+        bruteforcing attack of the SPNs account NTLM hash if we can gather valid TGS for those SPNs.
+        This is part of the kerberoast attack research by Tim Medin (@timmedin).
+    ''',
+    'authors': [
+        'Alberto Solino', # impacket example
+        'Jacob Robles'    # Metasploit module conversion
+    ],
+    'date': '2014-09-27',
+    'license': 'CORE_LICENSE',
+    'references': [
+        {'type': 'url', 'ref': 'https://github.com/CoreSecurity/impacket/blob/master/examples/GetUserSPNs.py'},
+        {'type': 'url', 'ref': 'https://files.sans.org/summit/hackfest2014/PDFs/Kicking%20the%20Guard%20Dog%20of%20Hades%20-%20Attacking%20Microsoft%20Kerberos%20%20-%20Tim%20Medin(1).pdf'},
+        {'type': 'aka', 'ref': 'GetUserSPNs.py'},
+        {'type': 'aka', 'ref': 'Kerberoast'}
+    ],
+    'type': 'scanner.single',
```

Comment:
Per https://github.com/rapid7/metasploit-framework/pull/9733, it should be single_scanner